src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -484,6 +484,11 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
 
             switch (intrinEmbMask.numOperands)
             {
+                case 4:
+                    assert(intrinEmbMask.op4 != nullptr);
+                    assert(HWIntrinsicInfo::HasImmediateOperand(intrinEmbMask.id));
+                    FALLTHROUGH;
```

Comment:
can you also save `embMaskOp4Reg` and then later, check if it is `REG_NA` because it was contained?

No need for `assert(HWIntrinsicInfo::HasImmediateOperand(intrinEmbMask.id));` at this point.